modin/pandas/groupby.py

```diff
@@ -921,16 +1138,49 @@ def _apply_agg_function(self, f, *args, **kwargs):
             agg_kwargs=kwargs,
             groupby_kwargs=self._kwargs,
             drop=self._drop,
+            selection=selection,
         )
         if self._idx_name is not None and self._as_index:
             new_manager.set_index_name(self._idx_name)
         result = type(self._df)(query_compiler=new_manager)
+        self._verify_aggregation_result(result, f)
+
         if result._query_compiler.get_index_name() == "__reduced__":
             result._query_compiler.set_index_name(None)
         if self._squeeze:
-            return result.squeeze()
+            return result.squeeze(axis=1)
         return result
 
+    def _verify_aggregation_result(self, result, agg_func):
```

Comment:
this logic was moved from the [pandas backend](https://github.com/modin-project/modin/blob/b8098bd929c9971e5e5130e1605ff514dba69777/modin/backends/pandas/query_compiler.py#L2854-L2867) to be reused for any backend